lib/tasks/plugin_assets.rake

```diff
@@ -15,6 +15,10 @@ task 'plugin:assets:precompile', [:plugin] => [:environment] do |t, args|
         app = Rails.application
         app.config.assets.digest = true
         app.config.assets.precompile = plugin.assets
+        if plugin.gettext_domain
+          locale_assets = FastGettext.default_available_locales.map { |loc| "locale/#{loc}/#{plugin.gettext_domain}.js" }
```

Comment:
Is this really correct? I did some digging a while back and https://gist.github.com/ekohl/2927caf52ebb7a1c19bf84699f0fd934 is what I came up with. Short story:

* `locale/$lang/$domain.po` are the raw files
* `locale/$lang/LC_MESSAGES/$domain.mo` exists for Ruby processes
* `public/assets/locale/$lang/app.js` is for frontend

And `public/assets` should already be in the list of assets to precompile.

Reply:
I think it is. `public/assets/locale/$lang/$domain-$hash.js` is the precompiled variant
- `plugin:gettext` takes plugin sources and extracts strings into `$plugin_root/locale/**/*.po`
- `plugin:po_to_json` takes the `*.po` files and converts them to `*.js` for frontend, these end up in `app/assets/javascripts/locales/**/$domain.js`
- `plugin:assets:precompile` takes (among other things) the `*.js` files and puts them into `$plugin/public/assets/locale/$lang/$domain-$hash.js`

If I remove this part I don't get any `$domain-$hash.js` and in the logs I can see this
```
Plugin foreman-tasks has assets outside of its namespace, these will be ignored: locale/en/foreman_tasks.js, locale/fr/foreman_tasks.js, locale/ja/foreman_tasks.js, locale/zh_CN/foreman_tasks.js, locale/de/foreman_tasks.js, locale/es/foreman_tasks.js,
 locale/ka/foreman_tasks.js, locale/ko/foreman_tasks.js, locale/pt_BR/foreman_tasks.js, locale/ru/foreman_tasks.js, locale/zh_TW/foreman_tasks.js, locale/ca/foreman_tasks.js, locale/cs_CZ/foreman_tasks.js, locale/en_GB/foreman_tasks.js, locale/it/foreman_tasks.js, locale/pl/forem
an_tasks.js
```

Reply:
It seems to ignore the files that are missing, but changed it nevertheless